src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,22 +126,33 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return this.block.getBlock() + " " + this.street.getStreet() + " " +
+    this.unit.getUnit() + " " + this.postalCode.getPostal();
     }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+        if(other == this) // short circuit if same object
+        	return true;
+        else if(other instanceof Address) { // instanceof handles nulls
+        	Address address2 = (Address) other;
+        	return this.block.getBlock() == address2.block.getBlock() && 
+        			this.street.getStreet().compareTo(address2.street.getStreet()) == 0 &&
+        			this.unit.getUnit().compareTo(address2.unit.getUnit()) == 0 &&
+        			this.postalCode.getPostal() == address2.postalCode.getPostal();
+        }
+        else
```

Comment:
you can short circuit the else-case by writing ahead something like this:

    if (!(other instanceof Address)) {
        return false
    }

that will save one indentation for lines 138 to 142.